checkov/cloudformation/graph_builder/graph_to_definitions.py

```diff
@@ -20,9 +20,13 @@ def convert_graph_vertices_to_definitions(
         block_name = vertex.name.split('.')[-1]  # vertex.name is "type.name" so type.name -> [type, name]
 
         definition = {
-            'Type': vertex.attributes['resource_type'] if vertex.block_type == BlockType.RESOURCE else vertex.block_type,
-            'Properties': vertex.config
+            "Type": vertex.attributes["resource_type"] if vertex.block_type == BlockType.RESOURCE else vertex.block_type,
+            "Properties": vertex.config,
         }
+
+        if vertex.metadata:
+            definition["Metadata"] = vertex.metadata
```

Comment:
@gruebel Why not set it inside the `definition = {...`?

Reply:
because it is optional, not every resource has a `Metadata` section. It would be misleading, if I add it with an empty `dict` or `None` value